src/IotTelemetrySimulator/SimulatedDevice.cs

```diff
@@ -35,15 +36,21 @@ async Task RunnerAsync(RunnerStats stats, CancellationToken cancellationToken)
                 stats.IncrementDeviceConnected();
 
                 // Delay first event by a random amount to avoid bursts
-                await Task.Delay(this.random.Next(this.interval), cancellationToken);
+                int crtInterval = this.interval[0];
+
+                await Task.Delay(this.random.Next(crtInterval), cancellationToken);
 
                 var stopwatch = new Stopwatch();
                 stopwatch.Start();
+                ulong totalIntervalTime = 0;
                 for (var i = 0L; !cancellationToken.IsCancellationRequested && (this.config.MessageCount <= 0 || i < this.config.MessageCount); i++)
```

Comment:
Should we make this counter a ulong?
When we pass the max it will become negative (https://dotnetfiddle.net/VEQc6C).
This is when we make it unsigned: https://dotnetfiddle.net/AOu4DF

Reply:
I've removed this counter as it's not needed if we adjust only based on the last message.

Reply:
I have made all the counters long in the end and I suggest to reset them after a number of loops.